datadriven/src/line_sparser.rs

```diff
@@ -0,0 +1,136 @@
+use crate::test_data::CmdArg;
+use anyhow::Result;
+use regex::Regex;
+
+// Token
+// (1) argument (no value)
+// (2) argument= (no value)
+// (3) argument=() (no value)
+// (4) argument=a (single value)
+// (5) argument=a,b,c (multiple value)
+// (6) argument=(a,b,c,...) (multiple value)
+//
+// parse_line parses a line of datadriven input language and returns
+// the parsed command and CmdArgs.
+pub fn parse_line(line: &str, logger: &slog::Logger) -> Result<(String, Vec<CmdArg>)> {
+    let fields = split_directives(line)?;
+    if fields.is_empty() {
+        bail!("empty lines occurs, unexpected.");
+    }
+
+    debug!(logger, "argument after split: {:?}", fields);
+
+    let cmd = fields[0].clone();
+    let mut cmd_args = vec![];
+
+    for arg in &fields[1..] {
+        let key_value = arg
+            .replacen("=", " ", 1)
+            .split_terminator(' ')
+            .map(|v| v.to_string())
```

Comment:
In rust, we tend to allocate only when it's necessary. From what I can see, following code can happily accept `&str` and iterator` instead of `String` and `Vec`.